crates/hud/src/universal_ui.rs

```diff
@@ -0,0 +1,91 @@
+use bevy::asset::Handle;
+use bevy::prelude::{ButtonBundle, Color, NodeBundle, TextBundle};
+use bevy::text::{Font, HorizontalAlign, Text, TextAlignment, TextStyle, VerticalAlign};
+use bevy::ui::{AlignItems, AlignSelf, BackgroundColor, FlexDirection, JustifyContent, Size, Style, UiRect, Val};
+use bevy::utils::default;
+
+pub fn create_side_node_bundle(color: Color) -> NodeBundle {
```

Comment:
Ditto, please have a look at ui crate which is designed to implement all the low level UI building blocks.